v6_carrier_py/master.py

```diff
@@ -105,12 +112,53 @@ def correlation_matrix(client: ClientContainerProtocol, data, keys=None, *args,
     If no keys are specified the datasets are joined on all columns with the same name.
     TODO: What if different datasets use different keys to mean the same thing? How do we specify this?
     """
-    results = _dispatch_tasks(client, data, *args, method='get_data', **kwargs)
+    combined_df = _combine_all_node_data(client, data, keys, *args, **kwargs)
+
+    return combined_df.corr()
+
+
+def train_model(client: ClientContainerProtocol, data, pipeline: Pipeline, features: List[str], target: str, keys=None,
+                *args, **kwargs):
+    """
+    Retrieve data from nodes and train data analysis pipeline on it. Returns the performance of the resulting model.
+    TODO: How and where do we save our model?
+    """
+    try:
+        info(f'Training pipeline with the following steps: {pipeline.named_steps}')
+        results = _combine_all_node_data(client, data, keys, *args, **kwargs)
+
+        X = results[features].values
+        y = results[target].values
+
+        # Split data
+        X_train, X_test, y_train, y_test = train_test_split(X, y)
```

Comment:
This should at least use a fixed random state to enforce having the same split. 
And preferably you would want to have a dedicated test set, I guess that should be something that the data station itself specifies, but in practice you will probably also need information from other data stations. Not sure... Maybe you can add a comment to think about this?

Reply:
I guess a user would also want to have control over this...

Reply:
I will add a comment to think about this. I don't think the data station should necessarily already have split the data in train and test, the data is more generic than that.

I agree that the user would like to have control over this, but this can come at a later stage.

Reply:
I will add a random seed though